src/libnpf/npf.c

```diff
@@ -1061,6 +1061,34 @@ npf_table_insert(nl_config_t *ncf, nl_table_t *tl)
 	return 0;
 }
 
+int
+npf_table_replace(nl_table_t *tl, int fd, npf_error_t *errinfo)
+{
+	nvlist_t *errnv = NULL;
+	int error;
+
+	/* Ensure the table is built. */
+	(void)_npf_table_build(tl);
+
+	if (nvlist_xfer_ioctl(fd, IOC_NPF_TABLE_REPLACE, tl->table_dict, &errnv) == -1) {
+		assert(errnv == NULL);
+		return errno;
+	}
+	error = dnvlist_get_number(errnv, "errno", 0);
+	if (error && errinfo) {
+		memset(errinfo, 0, sizeof(npf_error_t));
```

Comment:
This is duplicated with the code in `npf_config_submit()`. Can you please share it by adding a new function, e.g. `_npf_extract_error()`?

Reply:
Done locally & will push with other changes.